sync_diff_inspector/splitter/random.go

```diff
@@ -41,6 +41,17 @@ type RandomIterator struct {
 	dbConn *sql.DB
 }
 
+// MockRowCount is the mock row count used in unit tests
+var MockRowCount int64
+
+// a wrapper for get row count to integrate failpoint.
+func getRowCount(ctx context.Context, db dbutil.QueryExecutor, schemaName string, tableName string, where string, args []any) (int64, error) {
+	failpoint.Inject("getRowCount", func() {
```

Comment:
you can pass the injected value as the argument of failpoint. No need to use `MockRowCount`. You can find this usage in many failpoint inject in tidb repo.